scipy/linalg/_matfuncs.py

```diff
@@ -302,9 +303,37 @@ def expm(A):
     elif a.dtype == np.float16:
         a = a.astype(np.float32)
 
-    # An explicit formula for 2x2 case exists (formula (2.2) in [1]). However, without
-    # Kahan's method, numerical instabilities can occur (See gh-19584). Hence removed
-    # here until we have a more stable implementation.
+    # Explicit formula for 2x2 case (formula (2.2) in [1]).
+    if a.shape[-2:] == (2, 2): 
+        norm = np.max(np.linalg.norm(a).astype(int))
```

Comment:
This needs to be a bit more complicated I'm afraid. 

But just to start from this line. Not sure what int is doing there but we need to get each 2x2 array treated separately. 

Say you have a 5x2x2 arrays. We need to treat 5 different 2x2 arrays. And scale them independently. If you want to scale you can use the closest power of 2 (including negative powers) such that scaling is not noisy since powers of two would be an exact representation.